lib/Connector/AlphaVantageConnector.php

```diff
@@ -255,16 +255,19 @@ private function getStockResults(DataProviderInterface $dataProvider): void
 
                 $item = [];
 
-                foreach ($result['Time Series (Daily)'] as $series) {
+                $isFromCache = !array_key_exists('Time Series (Daily)', $result);
```

Comment:
This doesn't make sense - the `getStockQuote()` should always return data in the same format/structure.

Reply:
Yes, supposedly the cache should always return the same format/structure but apparently in this case, it's not. Upon checking the format/structure that is being saved into the cache is the formatted output already.

```

                foreach ($results as $series) {
                    $item = [
                        'Name' => $name,
                        'Symbol' => $symbol,
                        'time' => $isFromCache ? $series['time'] : $result['Meta Data']['3. Last Refreshed'],
                        'LastTradePriceOnly' => $isFromCache ? $series['LastTradePriceOnly'] : round($series['4. close'], 4),
                        'RawLastTradePriceOnly' => $isFromCache ? $series['RawLastTradePriceOnly'] : $series['4. close'],
                        'YesterdayTradePriceOnly' => $isFromCache ? $series['YesterdayTradePriceOnly'] : round($series['1. open'], 4),
                        'RawYesterdayTradePriceOnly' => $isFromCache ? $series['RawYesterdayTradePriceOnly'] : $series['1. open'],
                        'TimeZone' => $isFromCache ? $series['TimeZone'] : $result['Meta Data']['5. Time Zone'],
                        'Currency' => $currency
                    ];

                    $item['Change'] = round($item['RawLastTradePriceOnly'] - $item['RawYesterdayTradePriceOnly'], 4);
                    $item['SymbolTrimmed'] = explode('.', $item['Symbol'])[0];
                    $item = $this->decorateWithReplacements($item);
                    break;
                }

```

Reply:
So the question in that case is why?

We are in control of saving to the cache, and returning from the cache, so we ought to be able to make sure they are the same?

Reply:
I'll try to clear my local cache for this module and see maybe the one I'm getting is an old cache? But yes, we should have control of what we want to save to the cache.